src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -3850,7 +3865,23 @@ internal override void WriteGeneratedEdmModel(string escapedEdmxString)
                 global::System.Xml.XmlReader reader = CreateXmlReader(Edmx);
                 try
                 {
-                    return global::Microsoft.OData.Edm.Csdl.CsdlReader.Parse(reader);
+                    global::Microsoft.OData.Edm.Csdl.CsdlReaderSettings edmxReaderSettings = new global::Microsoft.OData.Edm.Csdl.CsdlReaderSettings()
+                    {
+                        IgnoreUnexpectedAttributesAndElements = ");
+this.Write(this.context.IgnoreUnexpectedElementsAndAttributes.ToString().ToLower());
+this.Write(@"
+                    };
+                    
+                    global::System.Collections.Generic.IEnumerable<global::Microsoft.OData.Edm.Validation.EdmError> errors;
+                    global::System.Collections.Generic.IEnumerable<global::Microsoft.OData.Edm.IEdmModel> references = global::System.Linq.Enumerable.Empty<global::Microsoft.OData.Edm.IEdmModel>();
+                    global::Microsoft.OData.Edm.IEdmModel edmModel;
+                    
+                    if (!global::Microsoft.OData.Edm.Csdl.CsdlReader.TryParse(reader, references, edmxReaderSettings, out edmModel, out errors))
+                    {

```

Comment:
If all you are setting on readerSettings is IgnoreUnexpectedAttributesAndElements, why not use this overload, and not build readerSettings or references?:
       public static bool TryParse(XmlReader reader, bool ignoreUnexpectedAttributesAndElements, out IEdmModel model, out IEnumerable<EdmError> errors)

Reply:
Thanks for pointing that out! This greatly simplifies the code.